server/base/src/main/java/org/apache/accumulo/server/conf/TableConfiguration.java

```diff
@@ -213,6 +216,11 @@ private static ScanDispatcher createScanDispatcher(AccumuloConfiguration conf,
         conf.getAllPropertiesWithPrefixStripped(Property.TABLE_SCAN_DISPATCHER_OPTS);
 
     newDispatcher.init(new ScanDispatcher.InitParameters() {
+
+      // scan dispatcher are in the critical path for scans, so only create ServiceEnv if needed.
+      private final Supplier<ServiceEnvironment> senvSupplier =
+          Suppliers.memoize(() -> new ServiceEnvironmentImpl(context));
```

Comment:
Under which circumstances would this one not be used, and the lazy loading would help?

Reply:
The default scan dispatcher does not use the service env during init().  However this should only be called when config changes, so thinking that memoize is not needed here.

Reply:
I removed the call to memoize, don't think it adds any benefit here.